google_guest_agent/network/manager/netplan_linux.go

```diff
@@ -0,0 +1,384 @@
+	}
+	return true, nil
+}
+
+// SetupEthernetInterface sets the network interfaces for netplan by writing drop-in files to the specified
+// configuration directory.
+func (n netplan) SetupEthernetInterface(ctx context.Context, config *cfg.Sections, nics *Interfaces) error {
+	// Create a network configuration file with default configurations for each network interface.
+	googleInterfaces, googleIpv6Interfaces := interfaceListsIpv4Ipv6(nics.EthernetInterfaces)
+
+	// Write the config files.
+	if err := n.writeNetplanEthernetDropin(googleInterfaces, googleIpv6Interfaces); err != nil {
+		return fmt.Errorf("error writing network configs: %v", err)
+	}
+
+	// If we are running netplan+systemd-networkd we try to write networkd's drop-in for configs
+	// not mapped/supported by netplan.
+	if err := n.writeNetworkdDropin(googleInterfaces, googleIpv6Interfaces); err != nil {
+		return fmt.Errorf("error writing systemd-networkd's drop-in: %v", err)
+	}
+
+	osInfo := osinfoGet()
+	// Debian 12 has a pretty generic matching netplan configuration for gce, until we have that
+	// changed we are only removing.
+	if osInfo.OS == "debian" && osInfo.Version.Major == 12 {
+		if err := os.Remove("/etc/netplan/90-default.yaml"); err != nil {
+			logger.Debugf("Failed to remove default netplan config: %s", err)
+		}
+	}
+
+	// Avoid restarting systemd-networkd.
+	if err := run.Quiet(ctx, "netplan", "apply"); err != nil {
+		return fmt.Errorf("error applying netplan changes: %w", err)
+	}
+
+	return nil
+}
+
+// writeNetworkdDropin writes the overloading network-manager's drop-in file for the configurations
+// not supported by netplan.
+func (n netplan) writeNetworkdDropin(interfaces, ipv6Interfaces []string) error {
+	stat, err := os.Stat(n.networkdDropinDir)
+	if err != nil {
+		return fmt.Errorf("failed to stat systemd-networkd's drop-in root dir: %w", err)
+	}
+
+	if !stat.IsDir() {
+		return fmt.Errorf("systemd-networkd drop-in dir(%s) is not a dir", n.networkdDropinDir)
+	}
+
+	for i, iface := range interfaces {
+		logger.Errorf("writing systemd-networkd drop-in config for %s", iface)
+
+		// Create and setup ini file.
+		data := networkdNetplanDropin{
+			Network: systemdNetworkConfig{
+				DNSDefaultRoute: true,
+			},
+		}
+
+		// We are only interested on DHCP offered routes on the primary nic,
+		// ignore it for the secondary ones.
+		if i != 0 {
+			data.Network.DNSDefaultRoute = false
+			data.DHCPv4 = &systemdDHCPConfig{
+				RoutesToDNS: false,
+				RoutesToNTP: false,
+			}
+			data.DHCPv6 = &systemdDHCPConfig{
+				RoutesToDNS: false,
+				RoutesToNTP: false,
+			}
+		}
+
+		if err := data.write(n, iface); err != nil {
+			return fmt.Errorf("failed to write systemd drop-in config: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// networkdDropinFile returns an interface's netplan drop-in file path.
+func (n netplan) networkdDropinFile(iface string) string {
+	return path.Join(n.networkdDropinDir, fmt.Sprintf("10-netplan-%s.d", iface), "override.conf")
```

Comment:
can we add a comment saying why we're hardcoding higher priority (10)

Reply:
done. I also added a comment the value of the netplan's drop in priority.